cmd/containerboot/main.go

```diff
@@ -478,11 +478,6 @@ runLoop:
 					if egressIPsHaveChanged && len(egressAddrs) != 0 {
 						for _, egressAddr := range egressAddrs {
 							ea := egressAddr.Addr()
-							// TODO (irbekrm): make it work for IPv6 too.
-							if ea.Is6() {
```

Comment:
Here we will also need to check if the host supports IPv6 and supports IPv6NAT else this will error out if we attempt to set up the rules for an IPv6 egress endpoint. Calling this method should be sufficient for a check https://github.com/tailscale/tailscale/blob/main/util/linuxfw/nftables_runner.go#L529

I guess what we want to do is:
- if the egress endpoint returns IPs for both families and the host supports both, set up for both
- if the egress endpoint returns IPs for both families, but the host only supports IPv4, set up only IPv4
- if the egress endpoint only returns IPv6, but the host does not support IPv6 then I guess we want to either a) crash so users see that there is an issue b) log a warning and stay in the loop so that when/if the egress gets an IPv4 address, we can pick it up

Reply:
https://github.com/tailscale/tailscale/pull/12598 is related (I hit a panic whilst testing this PR on an IPv4-only-nftables host)

Reply:
Thanks for the catch! I think `log.Fatal` in that last condition makes sense and keeps it consistent with the surrounding logic. Added in 95008f51b90158f4637b50002a2891767ef45f3a, and also added a test in 2b03a731eeadb8b1655c8071863ae166fb8bc6d8. I tried it out for real in an IPv4-only cluster too, and it errors out crash-looping as expected.